source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_result_df.py

```diff
@@ -241,3 +246,46 @@ def test__final_sum_of_different_magnitudes_should_not_lose_precision(
 
     assert result_df.count() == 4
     assert result_df.where(col("Quantity") == "100000000000.001").count() == 4
+
+
+@pytest.mark.parametrize(
+    "quality_1, quality_2, quality_3, expected_quality",
+    [
+        (
+            TimeSeriesQuality.AsProvided.value,
+            TimeSeriesQuality.Estimated.value,
+            TimeSeriesQuality.Incomplete.value,
+            Quality.Incomplete.value,
+        ),
+        (
+            TimeSeriesQuality.AsProvided.value,
+            TimeSeriesQuality.Estimated.value,
+            TimeSeriesQuality.AsProvided.value,
+            Quality.Estimated.value,
+        ),
+        (
+            TimeSeriesQuality.AsProvided.value,
+            TimeSeriesQuality.AsProvided.value,
+            TimeSeriesQuality.AsProvided.value,
+            Quality.Measured.value,
+        ),
+    ],
+)
+def test__quality_is_lowest_common_denominator_among_Measured_estimated_and_incomplete(
+    enriched_time_series_factory, quality_1, quality_2, quality_3, expected_quality
+):
+
+    # quality in timeseries are maped as folow in the wholsale domain
+    # AsProvided (value 4) -> Measured (value 0)
+    # Estimated (value 3) -> Estimated (value 1)
+    # Incomplete (value 5) -> Incomplete (value 2)
+
+    df = (
+        enriched_time_series_factory(quality=quality_1, gridArea="805")
+        .union(enriched_time_series_factory(quality=quality_2, gridArea="805"))
+        .union(enriched_time_series_factory(quality=quality_3, gridArea="805"))

```

Comment:
Can we simplify to this?
```suggestion
        enriched_time_series_factory(quality=quality_1)
        .union(enriched_time_series_factory(quality=quality_2))
        .union(enriched_time_series_factory(quality=quality_3))

```

Reply:
I've optimized it.